src/mpid/ch3/include/mpid_coll.h

```diff
@@ -718,3 +718,382 @@ static inline int MPID_Iscatterv(const void *sendbuf, const int *sendcounts,
 
     return mpi_errno;
 }
+
+
+#undef FUNCNAME
+#define FUNCNAME MPID_Ibarrier_sched
+#undef FCNAME
+#define FCNAME MPL_QUOTE(FUNCNAME)
+static inline int MPID_Ibarrier_sched(MPIR_Comm * comm, MPIR_Sched_t s)
+{
+    int mpi_errno = MPI_SUCCESS;
+
+    mpi_errno = MPIR_Ibarrier_intra_sched(comm, s);
```

Comment:
Why are we using the `_intra_` versions of these collectives here? Shouldn't we just call `MPIR_Ibarrier_sched`?

Reply:
This is equivalent to the `coll_fns`.  I didn't want to change that functionality.